### PR TITLE
Us13 backend db auth

### PR DESCRIPTION
**Pull request by Zakari-B**
Added a DB to the back end and a very basic Auth system.
Integrated said Auth system in the front-end to communicate with the back-end and allow user login on the webapp.
Added placeholders to the Fav and Shopping pages, dependant on whether the user is logged in or not.
Modified some style elements.

### CODE REVIEW

File: backend/.env.sample [hunk 1]
```diff
@@ -3,3 +3,4 @@ DB_HOST=localhost
 DB_USER=notroot
 DB_PASSWORD=helloworld
 DB_NAME=mvc_express
+TOKEN_SECRET=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiO…Tk0fQ.4FjW
```
Comment (lchastanet) [inline backend/.env.sample:6]: Don't push online secret and db server credentials...

File: backend/src/app.js [hunk 2]
```diff
@@ -9,6 +8,14 @@ const app = express();
 
 app.use(express.json());
 app.use(express.static(path.join(__dirname, "../public")));
+app.use(function AllowSelf(req, res, next) {
+  res.header("Access-Control-Allow-Origin", "http://localhost:3000"); // update to match the domain you will make the request from
```
Comment (lchastanet) [inline backend/src/app.js:12]: Put a wildcard here please

File: backend/src/models/UsersManager.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+//     .promise()
+//     .query("INSERT INTO 'users' (name, email, password) VALUES (?, ?, ?)", [
+//       name,
+//       email,
+//       hash,
+//     ])
+//     .then(([result]) => {
+//       return { id: result.insertId, name, email };
+//     });
+// };
```
Comment (lchastanet) [inline backend/src/models/UsersManager.js:71]: Commented code here...

File: backend/src/router.js [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
+router.get("/users/list", auth, UsersController.listUsers);
+router.post("/users/getUser", auth, UsersController.selectUser);
+router.post("/signUp", UsersController.signUp);
+router.post("/signIn", UsersController.signIn);
+
+// router.get("/items", ItemController.browse);
+// router.get("/items/:id", ItemController.read);
+// router.put("/items/:id", ItemController.edit);
+// router.post("/items", ItemController.add);
+// router.delete("/items/:id", ItemController.delete);
```
Comment (lchastanet) [inline backend/src/router.js:17]: Commented code here...

**Review by lchastanet — CHANGES_REQUESTED** [review]
Really well done, just clean it a bit 😊

File: backend/.env.sample (hunk 1 above)
Comment (Zakari-B) [reply backend/.env.sample:6]: It's a completely random password hash used as an exemple, and the DB_NAME is the default name found in env.sample.

File: backend/src/models/UsersManager.js (hunk 3 above)
Comment (Zakari-B) [reply backend/src/models/UsersManager.js:71]: Woops, didn't see this one.

File: backend/src/router.js (hunk 4 above)
Comment (Zakari-B) [reply backend/src/router.js:17]: And completely forgot this one as well.

**Review by lchastanet — APPROVED** [review]
Nickel 👌🏻

**Pull request merged**